Add Namespace import based on TargetFramework and framework version and adjust VB defaults

basoundr opened the pull request:
Tagging @srivatsn @davkean @dotnet/project-system for review

srivatsn commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.VisualBasic.targets:
```diff
@@ -17,15 +17,18 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DefineConstants>$(DefineConstants),$(ImplicitFrameworkDefine)</DefineConstants>
   </PropertyGroup>
-  <ItemGroup>
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETFramework'">
```
The condition on TFM doesn't sound right. So netstandard libraries don't get these default imports?

srivatsn replied:
For that reason you probably also want a different property to disable the default imports.

srivatsn commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.VisualBasic.targets:
```diff
@@ -17,15 +17,18 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Import Include="System" />
     <Import Include="System.Collections" />
     <Import Include="System.Collections.Generic" />
     <Import Include="System.Data" />
```
I'd say just drop System.Data altogether since that's not in the default set of referneces for netstandard\netcore

srivatsn commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.VisualBasic.targets:
```diff
@@ -31,4 +31,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- This namespace is introduced in 4.0 Framework assemblies -->
     <Import Include="System.Threading.Tasks" Condition=" '$(_TargetFrameworkVersionWithoutV)' >= '4.0' "/>
   </ItemGroup>
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(_IsNETCoreOrNETStandard)' == 'true'">
```
We should use some new property instead of DisableImplicitFrameworkReferences - probably something like DisableImplicitImports. However for .NETFramework, we should check both properties.

livarcocc commented:
@MattGertz for approval.

dsplaisted commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.VisualBasic.props:
```diff
@@ -13,6 +13,14 @@ Copyright (c) .NET Foundation. All rights reserved.
+    <OptionCompare Condition=" '$(OptionCompare)' == '' ">Binary</OptionCompare>
+    <OptionStrict Condition=" '$(OptionStrict)' == '' ">Off</OptionStrict>
+    <OptionInfer Condition=" '$(OptionInfer)' == '' ">On</OptionInfer>
+    <DisableImplicitNamespaceImports Condition="'$(DisableImplicitNamespaceImports)'==''">$(DisableImplicitFrameworkReferences)</DisableImplicitNamespaceImports>
```
This line (setting `DisableImplicitNamespaceImports`) should probably go in the .targets instead of the .props.  People will set `DisableImplicitFrameworkReferences` in the body of their project, but if this line is here then that won't affect the default value of `DisableImplicitNamespaceImports`.

basoundr commented:
@livarcocc Is the branching plan confirmed for 15.2? Will there be a 1.1.0 branch for this PR to retarget?

livarcocc commented:
@basoundr yes. there will be one. I will create it tomorrow and will let you know.

basoundr committed:
Namespace import differs for Framework projects and Net Core & Net Standard projects

The conditions are added to make them work appropriately

Also the VB default properties are set, only if they have not been set already.

basoundr commented:
@dotnet-bot retest this please

basoundr commented:
@livarcocc I think there is a bunch of failing of testcases(8, I think. They could be flaky as well. Not sure.) in this branch. Those failures are unrelated to my change. Could you please have someone look into it?

srivatsn commented:
@dotnet-bot retest this please
